### PR TITLE
Rebase PR #308

### PR DESCRIPTION
**Pull request by TSC21**
@EliaTarasov I rebased #308 and made some changes so to to have the rotation [MAV_SENSOR_ORIENTATION](https://mavlink.io/en/messages/common.html#MAV_SENSOR_ORIENTATION) enum value calculated in the `gazebo_mavlink_interface` (since it's only Mavlink related). Though I see some strange behaviour where the `orientation` value is always jumping around between 0 and another value out of the enum range. Can you please verify this PR and check what may be wrong with the approach? The only thing I did was moving things around from what you did, so I didn't change the logic.

### CODE REVIEW

**Comment by TSC21** [comment]
I have my concerns with what's happening in https://github.com/PX4/sitl_gazebo/pull/339/files#diff-2e4efcaacbe9dc622071d2932136bee2R162-R177. Could you please review those and make sure that those make sense?

**Comment by TSC21** [comment]
Also take into consideration the following: if the lidar/sonar links point forward (`X` aligned), then the sensor needs to point downward (`-Z` aligned). That means that, by default, the sensor points downwards when the link rotation is **0,0,0**. So that means you need always to take into consideration that the sensor rotation is **+90deg** in pitch, which means that if you have the link with **0, -90, 0** orientation, the sensor is pointing **forward**, so the sensor orientation is **`MAV_SENSOR_ROTATION_NONE`**. If any of the sensor SDFs don't follow this assumption then we need to make sure they do.

**Comment by EliaTarasov** [comment]
Ok, it makes sense. I will try this way in the next couple of days. Thanks!

File: src/gazebo_mavlink_interface.cpp [hunk 1]
```diff
@@ -609,6 +609,29 @@ void GazeboMavlinkInterface::send_mavlink_message(const mavlink_message_t *messa
   }
 }
 
+template <class T>
+void GazeboMavlinkInterface::setSensorOrientation(ignition::math::Vector3d u_Xs, T sensor_msg) {
```
Comment (mrivi) [inline src/gazebo_mavlink_interface.cpp]: sensor_msg.orientation needs to be passes as a reference otherwise it's never set

**Review by mrivi — CHANGES_REQUESTED** [review]
If you make the reference change it works as expected

File: src/gazebo_mavlink_interface.cpp (hunk 1 above)
Comment (TSC21) [reply src/gazebo_mavlink_interface.cpp]: Thanks @mrivi! That solves the problem!

**Comment by TSC21** [comment]
@mrivi @EliaTarasov I was able to add a way of determining the sensor rotation to both lidar and sonar, even though it's handle in a different way. This has been tested with lidars and sonars in different directions and in different aircrafts with the expected result. I am bringing this in now. Please have it tested on your side as well and let me know how it goes.

**Pull request merged**